vue/components/TheSearchPage.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+    <div id="homeBody" class="row justify-content-center">
+        <div class="col-lg-8">
+            <h3 id="searchTitle" class="page-header mx-2" v-cloak>{{query ? "Search: " + query : "Search"}}</h3>
+            <form id="searchForm">
+                <input v-model="searchBox" id="searchQuery" style="flex:1" type="text" class="form-control mr-2" placeholder="Search for users, communities, and tags." />
+                <button v-on:click.prevent="searchSubmit" class="button">Search</button>
+            </form>
+            <div v-cloak v-if="firstPageFetched" class="compact-container" id="resultsContainer">
+                <template v-for="result in results">
+                    <div v-bind:key="result._id" class="content-box">
+                        <div class="row">
+                            <tagResult v-bind="result" v-if="result.type=='tag'"></tagResult>
+                            <userResult v-bind="result" v-else-if="result.type=='user'"></userResult>
+                            <communityResult v-bind="result" v-else-if="result.type=='community'"></communityResult>
+                        </div>
+                    </div>
+                </template>
+            </div>
+            <loadingSpinner v-if="loading || allResultsLoaded" v-bind:loading="loading" v-bind:message="loading ? '' : (firstPageFetched ? 'No more results.' : 'No results found, sorry.')" />
+        </div>
+    </div>
+</template>
+
+<script>
+import loadingSpinner from './SharedSubComponents/loadingSpinner.vue'
+import infiniteLoader from './SharedSubComponents/infiniteLoaderMixin.js'
+import tagResult from './SearchPageComponents/tagResult.vue'
+import userResult from './SearchPageComponents/userResult.vue'
+import communityResult from './SearchPageComponents/communityResult.vue'
+export default {
+  
+    components: {
+      loadingSpinner, tagResult, userResult, communityResult
+    },
+
+    mixins: [infiniteLoader],
+
+    data: function () { return {
+        query: '',
+        searchBox: ''
+    } },
+
+    // makes the initial request for results if we have a query from the url initially; uses the history api to change the query without reloading the page
+    // every time a new search is made
+    created: function() {
+        //TODO: replace with proper routing, someday
```

Comment:
Good todo. As we're not going to SPA in one leap, this is a good interim approach.